src/components/EditPartnership/EditPartnership.tsx

```diff
@@ -0,0 +1,115 @@
+import { Box, Button } from '@material-ui/core';
+import React, { FC } from 'react';
+import { Except } from 'type-fest';
+import { PartnershipType, UpdatePartnership } from '../../api';
+import { DialogForm, DialogFormProps } from '../Dialog/DialogForm';
+import { DialogState } from '../Dialog/useDialog';
+import {
+  CheckboxesField,
+  CheckboxOption,
+  RadioField,
+  RadioOption,
+  SubmitError,
+} from '../form';
+import { PartnershipCardFragment } from '../PartnershipCard/PartnershipCard.generated';
+import {
+  useDeletePartnershipMutation,
+  useUpdatePartnershipMutation,
+} from './EditPartnership.generated';
+
+interface EditPartnershipProps {
+  state: DialogState;
+  partnership: PartnershipCardFragment;
+}
+
+interface DialogInput {
+  partnership: Except<UpdatePartnership, 'id'>;
+}
+
+export const EditPartnership: FC<EditPartnershipProps> = ({
+  state,
+  partnership,
+}) => {
+  const [updatePartnership] = useUpdatePartnershipMutation();
+  const [deletePartnership] = useDeletePartnershipMutation();
+
+  const submit: DialogFormProps<DialogInput>['onSubmit'] = async (input) => {
+    const inputWithId = {
+      partnership: { ...input.partnership, id: partnership.id },
+    };
+    await updatePartnership({
+      variables: { input: inputWithId },
+      refetchQueries: ['ProjectPartnerships'],
+    });
+  };
+
+  const onDeleteClick = async () => {
+    const { id } = partnership;
+    await deletePartnership({
+      variables: { input: id },
+      refetchQueries: ['ProjectPartnerships'],
+    });
+    state.onClose();
```

Comment:
Hmm this is a new use case. We need to show this delete in progress state in the UI. This happens automatically with the form submit handler, but not with this handler. Let me think about the best way to handle this.

Reply:
Ok, I've commented out the onClose since with the prop fix above it has a TS error.  I added a TODO for you to come back to this.

Reply:
You need to pass the close reason right?

Reply:
Close reason and a `Form` object as second argument
<img width="918" alt="Screen Shot 2020-06-22 at 12 01 34 PM" src="https://user-images.githubusercontent.com/43487134/85336586-adf61800-b493-11ea-80b8-55975b89b92f.png">

Reply:
Fixed via custom submit action